citar.el

```diff
@@ -470,6 +473,71 @@ documentation for the return value and the meaning of
 REBUILD-CACHE and FILTER."
   (citar-select-ref :rebuild-cache rebuild-cache :multiple t :filter filter))
 
+(defun citar--sort-by-selection (selected-hash candidates)
+  "Sort the CANDIDATES by putting those in SELECTED-HASH first."
+  (let ((selected)
+        (others))
+    (dolist (cand candidates (nreverse (nconc others selected)))
+      (if (gethash (substring-no-properties cand) selected-hash)
+          (push cand selected)
+        (push cand others)))))
+
+(defun citar--multiple-completion-table (selected-hash candidates filter)
+  "Return a completion table for multiple selection.
+SELECTED-HASH is the hash-table containining selected-candidates.
+CANDIDATES is the list of completion candidates, FILTER is the function
+to filter them."
+  (citar--completion-table
+   candidates filter
+   `(display-sort-function . (lambda (cands) (citar--sort-by-selection ,selected-hash cands)))
+   `(group-function . (lambda (cand transform)
+                        (pcase (list (not (not transform))
+                                     (gethash (substring-no-properties cand) ,selected-hash))
+                          ('(nil nil) "Select Multiple")
+                          ('(nil t)   "Selected")
+                          ('(t   nil) cand)
+                          ('(t   t  ) (propertize cand 'face 'highlight)))))))
```

Comment:
OIC; here's the face. I guess this is an obvious one; are there are more subtle alternatives?

Reply:
> OIC; here's the face. I guess this is an obvious one; are there are more subtle alternatives?
> 
> EDIT: see here for some possible ideas.
> 
> https://github.com/minad/vertico/blob/crm/vertico-crm.el
> 
> This is the vertico-crm prototype, which is actually superior to the consult-crm one. It does appear to use the same face display by default, but it defines a `vertico-crm-selected` face, which we maybe don't want to do?

I think we should define a face, I just didn't think about it after using one that worked ok for me. I will do that.

Reply:
> EDIT: added a face but it inherits from highlight. If you suggest a different definition for it I will accept it.

No worry; that's bike-shedding we can worry about later. 

I merged this; thanks again!